plugins/optimization-detective/class-od-html-tag-processor.php

```diff
@@ -318,13 +343,21 @@ public function next_token(): bool {
 				$i = array_search( 'P', $this->open_stack_tags, true );
 				if ( false !== $i ) {
 					array_splice( $this->open_stack_tags, (int) $i );
+					array_splice( $this->open_stack_attributes, (int) $i );
 					array_splice( $this->open_stack_indices, count( $this->open_stack_tags ) + 1 );
 				}
 			}
 
 			$level                   = count( $this->open_stack_tags );
 			$this->open_stack_tags[] = $tag_name;
 
+			// For children of the BODY, capture disambiguating comments. See the get_xpath() method for where this data is used.
+			$attributes = array();
+			if ( isset( $this->open_stack_tags[1] ) && 'BODY' === $this->open_stack_tags[1] && 2 === $level ) {
+				$attributes = $this->get_disambiguating_attributes();
+			}
+			$this->open_stack_attributes[] = $attributes;
```

Comment:
Since this is not really aligned with what the `$open_stack_tags` property does (as this one only acts for the specific case of a `body` child element), maybe we should use another name? Could be something very specific like `body_child_element_attributes`.

Reply:
See https://github.com/WordPress/performance/pull/1797#discussion_r1926048532